src/webapi/blob.rs

```diff
@@ -31,6 +31,22 @@ pub trait IBlob: ReferenceType {
             Some( mime )
         }
     }
+
+    /// Create a new `Blob` object containing the data in the specified range of bytes of the
+    /// source `Blob`.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice)
+    /// https://w3c.github.io/FileAPI/#ref-for-dfn-slice
+    fn slice( &self, start: Option<u64>, end: Option<u64>, content_type: Option<&str> ) -> Blob {
```

Comment:
Hm... while this works it's not very Rusty.

Something like this using [RangeBounds](https://doc.rust-lang.org/std/ops/trait.RangeBounds.html) would be better I think:

```rust
fn slice< T >( &self, range: T ) -> Blob where T: RangeBounds { ... }
fn slice_with_content_type( &self, range: T, content_type: &str ) -> Blob where T: RangeBounds { ... }
```

then you could use it:

```
blob.slice( 0..12 );
blob.slice( ..12 );
blob.slice( 10.. );
// etc.
```